src/Tasks/WriteCodeFragment.cs

```diff
@@ -245,5 +274,312 @@ private string GenerateCode(out string extension)
             // as there's no point writing the file
             return haveGeneratedContent ? code : String.Empty;
         }
+
+        /// <summary>
+        /// Finds the metadata items that are used to indicate the types of
+        /// parameters, and removes those items from the given dictionary.
+        /// Returns a dictionary that maps parameter names to their declared types.
+        /// </summary>
+        private Dictionary<string, ParameterType> ExtractParameterTypes(IDictionary customMetadata)
+        {
+            Dictionary<string, ParameterType> parameterTypes = new();
+            List<string> keysToRemove = new();
```

Comment:
🤔 Looking at this again, it might be better to delay-create `keysToRemove`. The most common case will surely be not having any special keys, since that was the original usage. That means most of the time the list will be created for no reason.